scripts/irods/test/test_misc.py

```diff
@@ -471,3 +472,31 @@ def test_tcp_keepalive_probes__issue_2533_3824(self):
         # Check for SOCKET_ERROR (-179000) with an errno of 22 (Invalid argument).
         self.env['IRODS_TCP_KEEPALIVE_PROBES'] = str(2147483647)
         self.admin.assert_icommand('ils', 'STDERR', '-179022 SOCKET_ERROR, Invalid argument', env=self.env)
+
+
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', "Designed for the NREP")
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_enabling_and_disabling_of_msiSendMail__issue_7651(self):
+        config = IrodsConfig()
+        rep_name = f'{self.plugin_name}-instance'
+        rule = 'msiSendMail("x", "y", "z")'
+
+        # Show use of msiSendMail is blocked when the configuration option is not defined.
+        self.assertNotIn('enable_deprecated_msiSendMail', config.server_config['advanced_settings'])
+        self.user.assert_icommand(['irule', '-r', rep_name, rule, 'null', 'ruleExecOut'], 'STDERR', ['-169000 SYS_NOT_ALLOWED'])
+
+        # Show use of msiSendMail is blocked when the configuration option is set to false.
```

Comment:
please add another assertion when set to a non-True-or-False string, aka apples or bananas.

Reply:
Done.